contracts/permissionless/OrderbookReserve.sol

```diff
@@ -0,0 +1,870 @@
+        bool addedWithHint = false;
+
+        OrderListInterface list = isEthToToken ? ethToTokenList : tokenToEthList;
+
+        if (hintPrevOrder != 0) {
+            addedWithHint = list.addAfterId(maker, newId, srcAmount, dstAmount, hintPrevOrder);
+        }
+
+        if (addedWithHint == false) {
+            list.add(maker, newId, srcAmount, dstAmount);
+        }
+
+        NewLimitOrder(maker, newId, isEthToToken, srcAmount, dstAmount, addedWithHint);
+
+        return true;
+    }
+
+    event OrderUpdated(
+        address indexed maker,
+        bool isEthToToken,
+        uint orderId,
+        uint128 srcAmount,
+        uint128 dstAmount,
+        bool updatedWithHint
+    );
+
+    function updateOrder(address maker, bool isEthToToken, uint32 orderId, uint128 newSrcAmount,
+        uint128 newDstAmount, uint32 hintPrevOrder)
+        internal
+        returns(bool)
+    {
+        require(newSrcAmount < MAX_QTY);
+        require(newDstAmount < MAX_QTY);
+        uint128 currDstAmount;
+        uint128 currSrcAmount;
+        address orderMaker;
+
+        OrderListInterface list = isEthToToken ? ethToTokenList : tokenToEthList;
+
+        (orderMaker, currSrcAmount, currDstAmount,  ,  ) = list.getOrderDetails(orderId);
+        require(orderMaker == maker);
+
+        if (!validateUpdateOrder(maker, isEthToToken, currSrcAmount, currDstAmount, newSrcAmount, newDstAmount)) {
+            return false;
+        }
+
+        bool updatedWithHint = false;
+
+        if (hintPrevOrder != 0) {
+            (updatedWithHint, ) = list.updateWithPositionHint(orderId, newSrcAmount, newDstAmount, hintPrevOrder);
+        }
+
+        if (!updatedWithHint) {
+            list.update(orderId, newSrcAmount, newDstAmount);
+        }
+
+        OrderUpdated(maker, isEthToToken, orderId, newSrcAmount, newDstAmount, updatedWithHint);
+
+        return true;
+    }
+
+    event OrderCanceled(address indexed maker, bool isEthToToken, uint32 orderId, uint128 srcAmount, uint dstAmount);
+
+    function cancelOrder(bool isEthToToken, uint32 orderId) internal returns(bool) {
+
+        address maker = msg.sender;
+        OrderListInterface list = isEthToToken ? ethToTokenList : tokenToEthList;
+        OrderData memory orderData = getOrderData(list, orderId);
+
+        require(orderData.maker == maker);
+
+        uint weiAmount = isEthToToken ? orderData.srcAmount : orderData.dstAmount;
+
+        require(handleOrderStakes(orderData.maker, uint(calcKncStake(int(weiAmount))), 0));
+
+        require(removeOrder(list, maker, isEthToToken ? ETH_TOKEN_ADDRESS : token, orderId));
+
+        //funds go back to makers account
+        makerFunds[maker][isEthToToken ? ETH_TOKEN_ADDRESS : token] += orderData.srcAmount;
+
+        OrderCanceled(maker, isEthToToken, orderId, orderData.srcAmount, orderData.dstAmount);
+
+        return true;
+    }
+
+    ///@param maker is the maker of this order
```

Comment:
Some of the functions have this @ documentation and some do not. I think better be consistent.

Reply:
well.
its consistent with rest of our code :)

where is it missing?